src/averageAges.js

```diff
@@ -15,11 +16,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = century
+    ? people.filter((p) => p.sex === 'm' && Math.ceil(p.died / 100) === century)
+    : people.filter((p) => p.sex === 'm');
```

Comment:
It might be more readable to use a regular if-else statement instead of the ternary operator to make the condition easier to understand. Currently you are not handling a case where there is 0 men which leads to possible division by 0 in the last line.  Instead of iterating over men and using let for sum variable we could use [reduce array method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce?retiredLocale=pl). Code is also not formatted correctly in some places so you should consider using formatting tools like [prettier ](https://prettier.io/)
```js
function calculateMenAverageAge(people, century) {
  let men = [];

  if (century) {
    men = people.filter((p) => p.sex === 'm' && Math.ceil(p.died / 100) === century);
  } else {
    men = people.filter((p) => p.sex === 'm');
  }

  if (men.length === 0) {
    return 0; // Handle division by zero
  }

  const sum = men.reduce((acc, m) => acc + (m.died - m.born), 0);

  return sum / men.length;
}
```